src/test/java/de/telran/onlineshopforhomeandgarden1/service/ProductServiceTest.java

```diff
@@ -51,4 +55,27 @@ public void getProductByIdIncorrectTest() {
         Mockito.verify(repository).findById(111L);
         assertTrue(optional.isEmpty());
     }
+
+
+    @Test
+    public void getAllTest() {
+        Product product = new Product();
+        Pageable pageable = PageRequest.of(0, 10, Sort.by("name").ascending());
+        Long categoryId = null;
+        Boolean hasDiscount = null;
+        Integer minPrice = null;
+        Integer maxPrice = null;
+        List<Product> productList = Arrays.asList(product);
+        Page<Product> products = new PageImpl<>(productList);
+//        Page<Product> products = new PageImpl<>(Collections.singletonList(product));
+        ProductWithDiscountPriceResponseDto responseDto = new ProductWithDiscountPriceResponseDto();
+//        productMapper = Mockito.mock(ProductMapper.class);
+
+        Mockito.when(repository.getAllWithFilters(categoryId, null, false, BigDecimal.valueOf(0), BigDecimal.valueOf(Integer.MAX_VALUE), pageable))
```

Comment:
Mock expects  specific arguments and the service passes other arguments for example hasdiscount you pass as null and the service initialises them as false.